torch/csrc/Exceptions.cpp

```diff
@@ -113,6 +113,18 @@ could not be completed because the input matrix is singular.",
       PyModule_AddObject(
           module, "_DistStoreError", THPException_DistStoreError) == 0);
 
+  // NOLINTNEXTLINE(bugprone-assignment-in-if-condition)
+  ASSERT_TRUE(
+      THPException_DistQueueEmptyError = PyErr_NewExceptionWithDoc(
+          "torch.distributed.QueueEmptyError",
+          "Exception raised when an error occurs in the distributed store",
+          THPException_DistStoreError,
+          nullptr));
+  ASSERT_TRUE(
+      PyModule_AddObject(
+          module, "_DistQueueEmptyError", THPException_DistQueueEmptyError) ==
+      0);
```

Comment:
Is the purpose of doing this to add the exception into PyTorch python object exception?

Reply:
Yes -- we need to do this to surface it to Python. Though, honestly maybe we should move all of the PTD errors to use pybind instead of THP for exception translation. THP is just so painful to work with